### ISSUE TITLE
avahi busted on my asuswrt-merlin

### ISSUE THREAD

**Issue by privacyguy123** [Entware/Entware#989]
**Describe the bug**
Same as https://github.com/Entware/Entware/issues/483 but I didn't want to necrobump. I don't understand how to fix it ... ELI5

**Environment (please complete the following information):**
 - Entware feed: ?
 - Device RT-AX58U
 - Firmware version asuswrt-merlin  3004.388.4

```
RT-AX58U-5468:/jffs/configs# /opt/etc/init.d/rc.unslung check
 Checking dbus-daemon...              dead.
 Checking avahi-daemon...              dead.
 Checking AdGuardHome...              alive.
```
```
RT-AX58U-5468:/tmp/mnt/USB/entware/bin#  /opt/sbin/avahi-daemon
Failed to find group 'nogroup'.
```

Further reading suggests dbus is looking for a "root" user which does not exist on this router/it is not possible to set my login name to root specifically. I can manually add "nogroup" to ```/etc/group``` but that feels broken.

**Comment by zyxmon** [comment]
Just create nogroup group on your device. This is device specific.

**Comment by privacyguy123** [comment]
> Just create nogroup group on your device. This is device specific.

I've tried that and it's still busted. All avahi-tools complain the daemon isn't running. It is.

More stuff that makes no sense to me

```
RT-AX58U-5468:/tmp/home/root# strings `which avahi-daemon` | grep run/
/opt/var/run/avahi-daemon//pid
/opt/var/run/avahi-daemon/
mkdir("/opt/var/run/avahi-daemon/"): %s
Failed to create runtime directory /opt/var/run/avahi-daemon/.
/opt/var/run/avahi-daemon/socket
```

--debug shows lots of this after adding nogroup manually

```
sendmsg() to 224.0.0.251 failed: Required key not available
sendmsg() to 224.0.0.251 failed: Required key not available
sendmsg() to 224.0.0.251 failed: Required key not available
```

**Comment by privacyguy123** [comment]
Sorry this isn't a sufficient answer.

How can I get avahi to recognize it's running as a daemon on asuswt-merlin? I've read your dbus dependency package is expecting to run as root and that user is not available to me, I can't even change my log in name to that. 

Adding "nogroup" manaully gets me a step further but all avahi-utils complain no daemon is running.

**Comment by zyxmon** [comment]
The best place to ask your questions is - https://www.snbforums.com/forums/asuswrt-merlin-addons.60/
We do not provide help for specific devices. AFAIK - no one from Entware team has asuswrt-merlin router.

**Comment by privacyguy123** [comment]
Nobody over there knows why it doesn't work either.
